### DISCUSSION TITLE
Is it possible to share the platform site pckages while using a local virtual environment?

### DISCUSSION THREAD

**Discussion by huxuan** [pdm-project/pdm#2917] [Q&A] [answered]
When working in dev containers, it is likely to share some fundamental dependencies. For example, when using the [NGC PyTorch Containers](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch), we prefer to share the pytorch and some other packages. In the meanwhile, we still want to keep local virtual environment especially when working on different projects.

I investigated a little bit about the global project related features, but it does not seems to meet the requirements. Just let me know if I missed anything. I also found there is a `--site-packages` option for `pdm run`, but we also need that for `install` and other commands.

This may bring extra challenages when locking dependencies, I am ok if there is any workaround for it.

BTW, `pipenv` has a similar feature with `--site-packages`: https://pipenv.pypa.io/en/latest/advanced.html#working-with-platform-provided-python-components

**Comment by huxuan** [comment]
It seems that there is a config "include-system-site-packages" [1] for venv, but there is no related documentation and it is also not shown in the `pdm config`.

[1] https://github.com/pdm-project/pdm/blob/f429652f856a61f59ce5013e3195a30fbe88ce62/src/pdm/models/venv.py#L74

**Comment by frostming** [comment, marked as answer]
That's a config of venv creation. In fact you can give abitrary venv arguments to `pdm venv create` command:

```
pdm venv create -- --system-site-packages
```

**Reply by huxuan** [reply]
Hi @frostming, thanks for the response.

Accroding to the hints you provide, I find https://github.com/pdm-project/pdm/issues/2287 that is really helpful. `pdm venv create python --system-site-packages` with `--update-reuse-installed` works for me. Seems the positional argument `python` is needed to refer to the default python.

Just some non-critical inconvenienve I found:

- `pdm list` does not differentiate the system and local site packages, the location field seems to be the place, but all of them are empty.
-  If I create the venv `--with-pip`, `pdm run pip list` will not show the packages installed by `pdm`.

Let me know if this is by design or anything I can do to help. Thanks for creating the great tool!

**Reply by frostming** [reply]
>  the location field seems to be the place

This field has value only if the dependency is installed from a local directory.

**Reply by frostming** [reply]
> * If I create the venv `--with-pip`, `pdm run pip list` will not show the packages installed by `pdm`.

Not true:

```
❯ pdm run pip list
Package      Version Editable project location
------------ ------- ----------------------------------------
blinker      1.8.2
click        8.1.7
Flask        3.0.3
itsdangerous 2.2.0
Jinja2       3.1.4
MarkupSafe   2.1.5
myproj       1.1.1   /Users/fming/wkspace/playground/test-pdm
pip          24.0
Werkzeug     3.0.3
```

**Reply by huxuan** [reply]
> This field has value only if the dependency is installed from a local directory.

OK, if it is by design, the differentiation would be a feature request then. :-)

> 
> Not true

Sorry, it seems that I have messed up with the system-level pip. I can confirm it works.

Thanks!
